src/algorithms/cave_cellular_automata.cpp

```diff
@@ -0,0 +1,43 @@
+#include <vector>
+#include "src/algorithms/cave_cellular_automata.h"
+
+CaveCellularAutomata::CaveCellularAutomata(int rock_percentage, int iterations, int neighbourhood_threshold, int neighbourhood_size) : Algorithm(std::string("CaveCellularAutomata")) {
+    this->rock_percentage = rock_percentage;
+    this->iterations = iterations;
+    this->neighbourhood_threshold = neighbourhood_threshold;
+    this->neighbourhood_size = neighbourhood_size;
+}
+
+int CaveCellularAutomata::count_rock_neighboors(const std::vector<std::vector<tile>>& board, int x, int y){
+    int rock_neighboors = 0;
+
+    for(size_t i = x - neighbourhood_size; i <= x + neighbourhood_size; ++i){
+        for(size_t j = y - neighbourhood_size; j <= y + neighbourhood_size; ++j){
+            if (board[i][j] == tile::rock)
+                ++rock_neighboors;
+        } 
+    }
+
+    return rock_neighboors;
+}
+
+void CaveCellularAutomata::run(Game& game) {
+  std::cout<< "Running " << get_name() << " algorithm" << std::endl; 
```

Comment:
I think it would be good to use some logger library in the future. Do you know any?

Reply:
I will integrate google logging library(GLOG) in the future. It allows programmist to set level of verbosity, so it will fit our needs

Reply:
Ok, I've added a task on jira

Reply:
And I would move printing the "Running algorithm" message to Generator::generate method. We won't have to repeat logging in every Algorithm child class